version: '3'

services:
  reverse-proxy:
    image: traefik:v2.9.10
    container_name: traefik
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    ports:
      - 80:80
      - 443:443
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./_config/traefik.yml:/etc/traefik/traefik.yml:ro
      - ./_config/config.yml:/etc/traefik/config.yml:ro
      - ./_certs:/etc/certs:ro
    networks:
      - red
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik=true"
      - "traefik.http.middlewares.latency-check.circuitbreaker.expression=LatencyAtQuantileMS(50.0) > 100"
     # - "traefik.http.routers.authme.service=otp"
     # - "traefik.http.routers.authme.rule=Host(`otp.com`)"
     # - "traefik.http.routers.authme.middlewares=circuitbreaker@docker"
      - "traefik.http.services.authme.loadbalancer.server.port=80"
      

networks:
  red:
    external: true